packages/typespec-client-generator-core/src/lib.ts

```diff
@@ -159,6 +159,12 @@ export const $lib = createTypeSpecLibrary({
         default: `Cannot pass an empty value to the @clientName decorator`,
       },
     },
+    "duplicate-name": {
+      severity: "error",
+      messages: {
+        default: paramMessage`Duplicate name: "${"name"}" in language scope: "${"scope"}"`,
```

Comment:
I think there is maybe a little more improvement to the message or target that could be done here. 

Trying this out in the [playground](https://cadlplayground.z22.web.core.windows.net/typespec-azure/prs/1119/?c=aW1wb3J0ICJAYXp1cmUtdG9vbHMvdHlwZXNwZWMtY2xpZW50LWdlbmVyYXRvci1jb3JlIjsKCnVzaW5nIEHENC5DxSVHyCQuQ29yZTsKCkBzZXJ2aWNlCm5hbWVzcGFjZSBDb250b3NvLldpZGdldE1hbmFnZXI7Cgptb2RlbCBGb28ge30KCkDGdk5hbWUoIkZvbyIpxyFCYXLEIQ%3D%3D&e=%40azure-tools%2Ftypespec-autorest&options=%7B%7D)

it is maybe a bit confusing when you look at `Foo` why it is a duplicate name.

Wondering if 
1. you should target the `@clientName` decorator that introduce the conflict when applicable instead of the type, this would make it much clearer specially when you have multiple. Could also make the message more targetted then `Client name  "${"name"}" is conflicting  in language scope: "${"scope"}"`
2. when targetting the other type that might have a conflict but doesn't have `@clientName` maybe could use a different message that would indicate that somewhere else you have a `@clientName`

Reply:
refined, throw with different error message separately for decorator and non-decorator.

Reply:
much nicer, i think this might have introduced a bug though on targetting the one with clientName